crates/provider/src/chain.rs

```diff
@@ -106,8 +112,77 @@ impl<T: Transport + Clone, N: Network> ChainStreamPoller<T, N> {
                     }
                 };
                 self.known_blocks.put(number, block);
+                if self.known_blocks.len() == BLOCK_CACHE_SIZE.get() {
```

Comment:
The implication here is that we will have to wait for the next value from `poller_task` while we technically know already that there are more blocks we can fetch. However, implementing it may be somewhat complicated and feels like a premature optimization.

Reply:
hmm, imo like this loop's logic is flawed because this allows iterating next_yield..tip before we start yielding blocks again

I consider this a hotfix, but ideally we can solve this my improving this stream impl.